packages/babel-generator/src/buffer.ts

```diff
@@ -262,7 +262,13 @@ export default class Buffer {
       // We mark the start of each line, which happens directly after this newline char
       // unless this is the last char.
       if (last < len) {
-        this._mark(++line, 0, identifierName, filename);
+        this._mark(
+          // When manually adding multi-line content (such as a comment), `line` will be `undefined`.
+          line != undefined ? ++line : line,
+          0,
+          identifierName,
+          filename,
+        );
```

Comment:
If this is a source-less segment (it doesn't have an original line/column), then we only need to call mark 0/1 times (which is done above the loop). That's because any additional lines generated by the comment will also be source-less, and source-less segments at the beginning of a line are discarded by `gen-mapping`.
```suggestion
       if (last < len && line !== undefined) {
        this._mark(++line, 0, identifierName, filename);
```